sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobAsyncClientBase.java

```diff
@@ -230,6 +230,9 @@ protected BlobAsyncClientBase(HttpPipeline pipeline, String url, BlobServiceVers
             throw LOGGER.logExceptionAsError(
                 new IllegalArgumentException("'snapshot' and 'versionId' cannot be used at the same time."));
         }
+        if (blobName == null) {
```

Comment:
Is an empty string blobName allowed? If not, we should check for that as well.

And were all other Storage SDKs checked for this possible issue as well?

Reply:
Yes, we should check for empty strings too as that will also result in an invalid path.

Reply:
Since empty strings are used when setting the root directory in datalake, empty strings are considered valid:
https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileSystemClient.java#L215